### PR TITLE
Raise macOS default deployment target

### PR DESCRIPTION
**Pull request by DarkDust**
Raise the default deployment target for macOS from 10.6 to 10.9. This gets rid of a warning in Xcode 12 with pods that did not explicitly specify a deployment target.

Closes CocoaPods/CocoaPods#9884

### CODE REVIEW

**Comment by segiddins** [comment]
This is a breaking change, since it will dirty everyone’s Podfile.lock that contains a pod that is using the default

**Comment by DarkDust** [comment]
What does that mean for me? Is there some documentation or other action by me required, or do I just need to wait for other devs/maintainers to chime in?

**Comment by dnkoutso** [comment]
@DarkDust it probably means we cant merge it until we ship a 2.0 version of CocoaPods. For your original issue just supply the version of macOS in your `Podfile` to fix the Xcode warning.

**Comment by DarkDust** [comment]
I did specify `platform :osx, '10.9'` in my `Podfile`, but that doesn't change the deployment target version of the pods that specify a platform but omit the version. Well, it's just a harmless warning so not that important. It can wait.

**Comment by alexito4** [comment]
Just FYI this is also happening with iOS projects since iOS 8.0 is no longer supported. What's the solution here? It's expected that Pods update and specify a version instead of relying on the defaults?

**Comment by Kaspik** [comment]
@dnkoutso @segiddins I'm not sure about the requirement "wait for 2.0.0 Cocoapods version" here.

CocoaPods currently set `4.3` for iOS or `10.6` for OS  (https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/podfile/dsl.rb#L569) - and if libraries are not providing the info, they will set these values. But these values make projects non-compilable. So *everyone* needs to add `post-install` hook to make their project compile.

If you would update these default values to `9.0` for iOS and 10.9 for macOS with next release of CocoaPods 1.11.0, what exactly is the breaking change?

- everyone with post hook won't see any difference
- everyone with library that specifies this won't see any difference
- everyone with library that doesn't specify this and without post hook... is probably using older CocoaPods anyways (so can't upgrade anyways)

So current default values don't work and don't make project compilable, right?
Can we please do (not only) this update and update default values that were set *more than 6 years ago*?

Thanks

**Comment by paulb777** [comment]
@Kaspik See discussions at https://github.com/CocoaPods/CocoaPods/pull/10070 and https://github.com/CocoaPods/CocoaPods/issues/9884

tldr: podspecs and Podfiles should be updated

**Comment by Kaspik** [comment]
I saw it (I even was part of the convo: https://github.com/CocoaPods/CocoaPods/issues/9884#issuecomment-651600153), but that doesn't change anything on my comment, right?

Either CocoaPods should not have `any` default value, and leave it up to post install hook / libraries as you suggest, or it should offer `valuable` default value. Right now it doesn't check any of these two checkboxes...

Another option would be set the `minimum target` based on `platform` `Podfile` definition, not `default value` from `CocoaPods` `Core`. But that's a bigger change than just bumping the minimum values.

**Comment by paulb777** [comment]
@Kaspik Sorry, on a closer read, I see your point, I was answering about a Podfile default versus a podspec default.

The potential breakage with changing the podspec default is much more theoretical - a pod using an iOS 8 API that is deprecated or deleted in iOS 9 and consumers still using Xcode 11.  However, it would still broken either way with Xcode 12.

**Comment by Kaspik** [comment]
Hmmm, yeah, theoretically yes... Is there any reason why not to come up with 2.0.0 instead of 1.11.0 then? Or what's the timeline for CocoaPods for upcoming months? This seems like a reasonable update sooner rather than later.
